src/service/document-info-impl.js

```diff
@@ -95,3 +105,43 @@ export class DocInfo {
 function getPageViewId(win) {
   return String(Math.floor(win.Math.random() * 10000));
 }
+
+/**
+ * Returns a map object of link tag relations in document head.
+ * Key is the link rel, value is a list of corresponding hrefs.
+ * @param {!Document} doc
+ * @return {!Object<string, string|!Array<string>>}
+ */
+function getLinkRels(doc) {
+  const linkRels = map();
+  if (doc.head) {
+    const links = doc.head.querySelectorAll('link[rel]');
+    for (let i = 0; i < links.length; i++) {
+      const link = links[i];
+      const href = link.href;
+      const rels = link.getAttribute('rel');
+      if (!rels || !href) {
+        continue;
+      }
+
+      rels.split(/\s+/).forEach(rel => {
+        for (let i = 0; i < filteredLinkRels.length; i++) {
+          if (rel.indexOf(filteredLinkRels[i]) != -1) {
+            return;
+          }
+        }
+
+        if (!linkRels[rel]) {
```

Comment:
You can assign the existing value to a variable and then avoid a few lookups of it further down.